src/MeshTransfer.f90

```diff
@@ -0,0 +1,1570 @@
+  ENDTYPE MeshTransfer_1DCart
+
+  TYPE,EXTENDS(MeshTransfer_1Dbase) :: MeshTransfer_1DCyl
+!
+!List of type-bound procedures
+    CONTAINS
+      !> @copybrief MeshTransfer::init_1DCyl
+      !> @copydetails MeshTransfer::init_1DCyl
+      PROCEDURE,PASS :: init => init_1DCyl
+  ENDTYPE MeshTransfer_1DCyl
+!
+!===============================================================================
+  CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief  initialization for the base class
+!> @param this  The mesh transfer object
+!> @param pList The parameter list which defines the input
+!>
+!> This routine processes the parameter list and sets data into the base class.
+!> Required and optional parameters include:
+!>   - Required:
+!>     * map_in - type of incoming map (POINT,VOLUME,CONTINUOUS)
+!>     * map_out - type of outgoing map (POINT,VOLUME,CONTINUOUS)
+!>     * moment_in - number of polynomial coefficients (only for map_in=CONTINUOUS)
+!>     * pointmesh_in - 1D array of point locations (only for map_in=POINT)
+!>     * volumemesh_in - 1D array of edges of volumes (only for map_in=VOLUME)
+!>     * moment_out - number of polynomial coefficients (only for map_out=CONTINUOUS)
+!>     * pointmesh_out - 1D array of point locations (only for map_out=POINT)
+!>     * volumemesh_out - 1D array of edges of volumes (only for map_out=VOLUME)
+!>   - Optional:
+!>     * poly_transfer - if present, defines the order of polynomial transfer that is used
+!>
+    SUBROUTINE init_1Dbase(this,pList)
+      CLASS(MeshTransfer_1Dbase),INTENT(INOUT) :: this
+      CLASS(ParamType),INTENT(IN) :: pList
+
+      INTEGER(SIK) :: i
+      TYPE(StringType) :: tmpstr
+
+      REQUIRE(pList%has('map_in'))
+      REQUIRE(pList%has('map_out'))
+
+      CALL pList%get('map_in',tmpstr)
+      SELECTCASE(CHAR(tmpstr))
+        CASE('CONTINUOUS')
+          this%MapType_in=MapType_CONTINUOUS
+        CASE('POINT')
+          this%MapType_in=MapType_POINT
+        CASE('VOLUME')
+          this%MapType_in=MapType_VOLUME
+        CASE DEFAULT
+          WRITE(*,*) "invalid incomming map type"
+          STOP -3
+      ENDSELECT
+      IF(this%MapType_in == MapType_CONTINUOUS) THEN
+        CALL pList%get('moments_in',this%nmesh_in)
+      ELSEIF(this%MapType_in == MapType_POINT) THEN
+        CALL pList%get('pointmesh_in',this%mesh_in)
+        this%nmesh_in=SIZE(this%mesh_in)
+      ELSEIF(this%MapType_in == MapType_VOLUME) THEN
+        CALL pList%get('volumemesh_in',this%mesh_in)
+        this%nmesh_in=SIZE(this%mesh_in)-1
+      ENDIF
+
+      CALL pList%get('map_out',tmpstr)
+      SELECTCASE(CHAR(tmpstr))
+        CASE('CONTINUOUS')
+          this%MapType_out=MapType_CONTINUOUS
+        CASE('POINT')
+          this%MapType_out=MapType_POINT
+        CASE('VOLUME')
+          this%MapType_out=MapType_VOLUME
+        CASE DEFAULT
+          WRITE(*,*) "invalid outgoing map type"
```

Comment:
exception handler here?

Reply:
I think DBC would be better (shouldn't happen in production, only development).

Reply:
Done :(